fix(chart): add option to disable the webhook

arthlr opened the pull request:
## Description

In my use case of telepresence, the mutating webhook is not needed. To stick to the kiss programming principle, I would like to add an option to disable the webhook.

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [x] I made sure to update `./CHANGELOG.yml`.
 - [x] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [x] I updated `DEVELOPING.md` with any special dev tricks I had to use to work on this code efficiently.
 - [x] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

arthlr committed:
feat(chart): add option to disable the webhook

Signed-off-by: Arthur Le Roux <aleroux@wiremind.io>

thallgren reviewed (changes requested):
This looks promising, and I like the idea of being able to completely disable the webhook.

I think there must be some additional logic in cmd/traffic/cmd/manager/manager.go that disables the agent-injector http-service from starting (line 139), and prevents the creation of the `injectorCertGetter` (line 118)  because it will otherwise make attempts to either watch or mount the mutator-webhook-tls secret, which is no longer generated.

Commit:
Let agentInjector.enable=false disable all agent-injector related things

Moves the `agentInjector.webhook.enable` to `agentInjector.enable` and
ensures that everything related to the agent injector service is
disabled when set to `false`.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Fix bad indentation in changelog entry.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Add tests for disabled agent injector.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Make itest TelepresenceHelmInstall return name of captured TM-log.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

thallgren commented:
@capuche2412 I didn't hear back from you so I decided to use your commit as the base for a more elaborated change that is in line with what I requested. Thanks for contributing and sparking this improvement.

Commit:
Don't lint traffic-manager/traffic-agent code on windows.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

arthlr commented:
This looks amazing ! I was busy and I haven't found the time to complete this PR. 
Thanks a lot for this improvement :+1:
